### ISSUE TITLE
Circle fill color change is not rendered

### ISSUE THREAD

**Issue by yaugenka** [openlayers/openlayers#7398]
When I call style.getImage().getFill().setColor() the color gets changed in the object with no errors but the change is not rendered.
Doing the same thing with text style works as expected style.getText().getFill().setColor()
See [this example fiddle](https://jsfiddle.net/gjvjawnh/)

Probably it is related to the difference between ol.style.Text and ol.style.Circle - the latter is based on ol.style.Image and does not have its own getCircle method in ol.style.Style but is rather called via getImage one.

So it looks like a bug and/or some kind of inconsistency which can be improved here.

**Comment by geuis** [comment]
I'm actually having a similar problem, but its with `style.getImage().getStroke().setWidth(10)`. If I do that and then console the stroke, the `width_` prop is updated, but the new width is not used when rendering.

**Comment by zZzimi** [comment]
Will you fix this "bug"?

**Comment by raiyni** [comment]
You have to make a new image or create an actual circle. 

https://jsfiddle.net/nnz7c2mL/

**Comment by zZzimi** [comment]
@raiyni Thank you your solution, but I have a lot of features, and my style is dynamic. I get parameters from the features (as example rotation). So if I always create a new image, then my performace will be bad. I know I should cache my style, but because every feature could have an own style (because of rotation example), it is not easy and maybe not the best solution. My plan was, create a style, and set the fill, stroke, image etc... I can change the fill/stroke, if I have a polygon or linestring, but I have problem with the point geometry.

**Comment by raiyni** [comment]
Because you are literally creating an image and thus can't update it. Whereas the linestrings and geometries aren't. Create a new image. You don't need to create a whole new style. 

```
style.setImage(new ol.style.Circle({
    radius: 5,
    fill: new ol.style.Fill({
      color: 'red'
    })
  }));

```

**Comment by yaugenka** [comment]
Would it make sense to add setFill method to ol.style.Circle?

**Comment by deguich** [comment]
I have the same problem :
> When I call style.getImage().getFill().setColor() the color gets changed in the object with no errors but the change is not rendered.

ol.style.Circle is a RegularShape so it should be possible to use getFill and setColor. Functions exists, they do their job but it produce no effects to the rendered Circle object. 

So, currently, you don't need to create a new Circle. You just need to ask for re-rendering the Circle. Changing radius will call the rendering method.

```javascript
var img = style.getImage();
img.getFill().setColor('red');
// Force circle rendering
img.setRadius(img.getRadius());
```

Events could be used to make a new rendering for all objects using the changed fill ...

**Comment by rwgaj** [comment]
> Changing radius will call the rendering method.

@deguich Oh my god, thank you so much :D

I was struggeling a whole day to make this work and I knew it had to be a bug.
Funny thing is, I also coded a `function` to change the Circle's `radius` and
it worked like expected. So I racked my brain about why it didn't work with the `color` xD
The solution you provided is so simple, but I'd never thought of that,
cause I'm not used to openlayer's internal logic (just getting started with the API).

I'm sure this already helped and will continue to help a lot of people.
The devs should add a `refresh` function or something like that ^^

Again, thank you so much :)
